File: Contract.Architecture.Frontends/Contract.Architecture.Web.Core/src/app/model/kundenstamm/kunden/dtos/i-kunde.ts
import { ApiKunde } from './api/api-kunde';

export interface IKunde {
    id: string;
    name: string;
    balance: number;
    bankId: string;
}

export class Kunde {
    public static fromApiKunde(apiKunde: ApiKunde): IKunde {
        if (apiKunde == null) {
            return null;
        }

        return {
            id: apiKunde.id,
            name: apiKunde.name,
            balance: apiKunde.balance,
            bankId: apiKunde.bankId,
        };
    }
}

File: Contract.Architecture.Backends/Contract.Architecture.Backend.Core/Contract/Logic/Modules/Kundenstamm/Kunden/DTOs/IKunde.cs
using System;

namespace Contract.Architecture.Backend.Core.Contract.Logic.Modules.Kundenstamm.Kunden
{
    public interface IKunde
    {
        Guid Id { get; set; }

        string Name { get; set; }

        int Balance { get; set; }

        Guid BankId { get; set; }
    }
}